[Feature] add test for config version feature

roycephan opened the pull request:
#### Status

READY

#### What's this PR do?

- [x] Test case for save config in persistent
- [x] Test case for get the lastest version of config
- [x] Test case for revert version
- [x] Test case for revert version, then add new version

roycephan committed:
cooler code

roycephan committed:
Test store config version in persistent (BoltDB)

- create test persistent db
- util test: create fake backend config, agent config
- test store config in persistent

roycephan committed:
test change version

roycephan committed:
clear log

hieunmce commented on backend/config/boltdb.go:
```diff
@@ -26,6 +26,10 @@ func (b boltImpl) Read() (*Config, error) {
 	cfg := &Config{}
 	err := b.db.View(func(tx *bolt.Tx) error {
 		bucket := tx.Bucket([]byte(b.bucket))
+		if bucket == nil {
```
We should use another way to initialize variable

hieunmce commented on backend/config/config.go:
```diff
@@ -145,7 +149,7 @@ func (c *Config) UpdateConfigFromAgent() error {
 	c.Version.Checksum = checksum
 	c.Version.SyncAt = time.Now()
-	c.Version.VersionNumber = c.Version.SyncAt.Unix()
+	c.Version.VersionNumber = c.Version.SyncAt.UnixNano()
```
We should use another efficient way to duplicate handle

hieunmce commented on backend/config/test/config_test.go:
```diff
@@ -0,0 +1,134 @@
+	backendConfig "github.com/dwarvesf/smithy/backend/config"
+)
+
+func TestStoreConfigInPersistent_1(t *testing.T) {
```
we should add a comment to explain test case because of unclear in context

hieunmce commented on backend/config/test/config_test.go:
```diff
@@ -0,0 +1,134 @@
+	})
+}
+
+func TestStoreConfigInPersistent_2(t *testing.T) {
```
we should add a comment to explain test case because of unclear in context

hieunmce commented on backend/config/test/config_test.go:
```diff
@@ -0,0 +1,134 @@
+	})
+}
+
+func TestRevertConfig(t *testing.T) {
```
we should add a comment to explain test case because of unclear in context

roycephan committed:
change code as reviewer

- change version number to ID
- change name variable
- add description

roycephan committed:
connection persistent

- remove open connection boltDB when read yaml
- open connection, close in each func in bolt
